packages/connect-form/src/hooks/use-connection-color.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback } from 'react';
+
+type ColorCode =
+  | 'color1'
+  | 'color2'
+  | 'color3'
+  | 'color4'
+  | 'color5'
+  | 'color6'
+  | 'color7'
+  | 'color8'
+  | 'color9'
+  | 'color10';
+
+const COLOR_CODES_TO_UI_COLORS_DARK_THEME_MAP: Record<ColorCode, string> = {
+  color1: '#00A35C', // green.dark1
+  color2: '#71F6BA', // green.light1
+  color3: '#016BF8', // blue.base
+  color4: '#0498EC', // blue.light1
+  color5: '#FFC010', // yellow.base
+  color6: '#EF5752', // red.light1
+  color7: '#B45AF2', // purple.base
+  color8: '#F1D4FD', // purple.light2
+  color9: '#889397', // gray.base
+  color10: '#C1C7C6', // gray.light1
+};
+
+const LEGACY_COLORS_TO_COLOR_CODE_MAP: Record<string, ColorCode> = {
+  '#5fc86e': 'color1',
+  '#326fde': 'color2',
+  '#deb342': 'color3',
+  '#d4366e': 'color4',
+  '#59c1e2': 'color5',
+  '#2c5f4a': 'color6',
+  '#d66531': 'color7',
+  '#773819': 'color8',
+  '#3b8196': 'color9',
+  '#ababab': 'color10',
+};
+
+export const CONNECTION_COLOR_CODES = Object.keys(
+  COLOR_CODES_TO_UI_COLORS_DARK_THEME_MAP
+) as ColorCode[];
+
+function isColorCode(hexOrColorCode: string | undefined) {
+  return hexOrColorCode
+    ? (CONNECTION_COLOR_CODES as string[]).includes(hexOrColorCode)
+    : undefined;
+}
```

Comment:
```suggestion
function isColorCode(hexOrColorCode: string | undefined): hexOrColorCode is ColorCode {
  return (CONNECTION_COLOR_CODES as string[]).includes(hexOrColorCode);
}
```

(renders the cast in the next function unnecessary)